adapter/pokemon_adapter.go

```diff
@@ -0,0 +1,62 @@
+package adapter
+
+import (
+	"fmt"
+	"strconv"
+
+	"github.com/nibble-4bits/ondemand-go-bootcamp/entity"
+)
+
+type pokemonAdapter struct {
+	dataSource DataSource
+	pokemons   []entity.Pokemon
+}
+
+func NewPokemonAdapter(ds DataSource) *pokemonAdapter {
+	adapter := &pokemonAdapter{dataSource: ds}
+
+	adapter.getPokemons()
+
+	return adapter
+}
+
+func (a *pokemonAdapter) getPokemons() {
+	csvRecords := a.dataSource.ReadCollection()
+
+	// Remove header from slice of records
+	csvRecords = csvRecords[1:]
+
+	for _, v := range csvRecords {
+		p := entity.Pokemon{}
+
+		p.Id, _ = strconv.Atoi(v[0])
```

Comment:
Normally, ignoring an error is not a good practice in Go. 

What happens if there's an error trying to convert a value? You need to have in mind the impact of that error not being handled.

I know it would be a pain to have the error checked for every single item, but you could create a function that tries to convert all ints (and another for the booleans), and handle the error there.

Something like this:
``` go
	 	 p.ID = parseToInt(v[0])
	 	 p.Total = parseToInt(v[1]) 
	 	 ...
	 	 
	func parseToInt(v string) int {
		num, err := strconv.Atoi(v)
		if err != nil {
			log.Warn("error converting x value... using default value 0")
		}

		return num
	}
```
You get the idea.

Reply:
You're right, I assumed that since the CSV is correct, there's no need to check for the parse error. But that's assuming the CSV (or any other data source, for that matter) is correct.

Now my question is, where should I put the `parseToInt` function? I was thinking of creating a `helpers` package, but I've read that `helpers` or `utils` package are not a good practice, because packages should have descriptive names and well-defined purposes